domain/tests/src/Unit/Access/DomainAccessCheckTest.php

```diff
@@ -0,0 +1,149 @@
+  }
+
+  /**
+   * @param string $path
+   *
+   * @dataProvider providerCheckPathFalse
+   */
+  public function testAppliesFalse($path) {
+    $route = $this->getMockBuilder('\Symfony\Component\Routing\Route')
+      ->disableOriginalConstructor()
+      ->getMock();
+    $route->expects(static::once())->method('getPath')
+      ->willReturn($path);
+    static::assertFalse($this->object->applies($route));
+  }
+
+  /**
+   * @param string $path
+   *
+   * @dataProvider providerCheckPathTrue
+   */
+  public function testCheckPathTrue($path) {
+    static::assertTrue($this->object->checkPath($path));
+  }
+
+  /**
+   * @param string $path
+   *
+   * @dataProvider providerCheckPathFalse
+   */
+  public function testCheckPathFalse($path) {
+    static::assertFalse($this->object->checkPath($path));
+  }
+
+  /**
+   * @todo Find a way to mock AccessResult::allowedIfHasPermissions to complete test
+   *
+   * If AccessResult::allowedIfHasPermissions cannot be done by mock, then
+   * it will need to be done as a web test case. Maybe an example in comment\Unit\Entity\CommentLockTest
+   */
+  public function testAccess() {
+    $domain = $this->getMockBuilder('\Drupal\domain\DomainInterface')
+      ->disableOriginalConstructor()->getMock();
+    $domain->expects(static::exactly(1))->method('status')
+      ->willReturnOnConsecutiveCalls(TRUE, FALSE, FALSE);
+    $this->mockNegotiator->expects(static::any())->method('getActiveDomain')
+      ->willReturnOnConsecutiveCalls(NULL, $domain, $domain, $domain);
+
+    $account = $this->getMockBuilder('\Drupal\Core\Session\AccountInterface')
+      ->disableOriginalConstructor()->getMock();
+    $account->expects(static::never())->method('hasPermission');
+
+    $expected = AccessResult::allowed()->setCacheMaxAge(0);
+    static::assertEquals($expected, $this->object->access($account));
+    static::assertEquals($expected, $this->object->access($account));
+
+
+/*    $manager = $this
+      ->getMockBuilder('\Drupal\Core\Cache\Context\CacheContextsManager')
+      ->disableOriginalConstructor()->getMock();
+    $container = new ContainerBuilder();
+    $container->set('cache_contexts_manager', $manager);
+
+    $account = $this->getMockBuilder('\Drupal\Core\Session\AccountInterface')
+      ->disableOriginalConstructor()->getMock();
+    $account->expects(static::exactly(1))->method('hasPermission')
+      ->willReturn(TRUE);
+    static::assertEquals($expected, $this->object->access($account));
+
+    return;
+    $account = $this->getMockBuilder('\Drupal\Core\Session\AccountInterface')
+      ->disableOriginalConstructor()
+      ->getMock();
+    $account->expects(static::exactly(2))->method('hasPermission')
+      ->willReturn(FALSE);
+    $expected = AccessResult::forbidden()->setCacheMaxAge(0);
+    static::assertEquals($expected, $this->object->access($account));*/
+  }
+
+  /**
+   * @return array
+   */
+  public function providerCheckPathTrue() {
+    return [
+      ['/user/1'],
```

Comment:
I wanted to make sure that the test cases captured places along the edge of expected behavior. So, if someone got and started pulling path from a different source, it would turn up. Moot issue now because the changes to the DomainAccessCheck that switched to to a list from config makes those tests entirely incorrect. I have updated the tests accordingly.